File: app/src/main/java/com/work/hany/playinseoul/main/MainActivity.java
package com.work.hany.playinseoul.main;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;

import com.work.hany.playinseoul.R;
import com.work.hany.playinseoul.tourlist.TourListActivity;
import com.work.hany.playinseoul.tourlist.TourListFragment;
import com.work.hany.playinseoul.util.ActivityUtils;

import javax.inject.Inject;

import dagger.Lazy;
import dagger.android.support.DaggerAppCompatActivity;

public class MainActivity extends DaggerAppCompatActivity {
//    private static final String CURRENT_FILTERING_KEY = "CURRENT_FILTERING_KEY";

    @Inject
    Lazy<MainFragment> mainFragmentProvider;


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        MainFragment mainFragment = (MainFragment) getSupportFragmentManager().findFragmentById(R.id.contentFrame);
        if (mainFragment == null) {
            mainFragment = mainFragmentProvider.get();
            ActivityUtils.addFragmentToActivity(getSupportFragmentManager(), mainFragment, R.id.contentFrame);
        }


        findViewById(R.id.search_icon_image_view).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, TourListActivity.class);
                startActivity(intent);
            }
        });
    }

}

File: app/src/main/java/com/work/hany/playinseoul/network/MainActivity.kt
package com.work.hany.playinseoul.network

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import com.work.hany.playinseoul.R
import com.work.hany.playinseoul.network.network.PlayInSeoulRetrofit
import com.work.hany.playinseoul.network.network.PlayInSeoulService
import com.work.hany.playinseoul.network.network.Result
import retrofit2.Call
import retrofit2.Callback

class MainActivity: AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        var service = PlayInSeoulRetrofit.instance.create(PlayInSeoulService::class.java)
        service.getAreaBasedList().enqueue(object : Callback<Result>{

            override fun onResponse(call: Call<Result>, response: retrofit2.Response<Result>?) {
                Log.d("HANY_TAG", "onResponse call.isCanceled ? " + call.isCanceled)
            }

            override fun onFailure(call: Call<Result>?, t: Throwable?) {
                Log.d("HANY_TAG", "onFailure ");
            }
        })
    }

}